.github/workflows/docker.yml

```
@@ -0,0 +1,68 @@
+name: Docker Build & Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: skynetlabs/skyd source branch (manual dispatch does NOT push to dockerhub)
+        type: string
+        required: true
+  pull_request:
+  push:
+    tags:
+      - v*
+
+jobs:
+  dockerfile-lint:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true # fail fast to prevent executing build-and-push
+      matrix:
+        flavor:
+          - bullseye-slim
+          - scratch
+    steps:
+      - uses: actions/checkout@v3
+      - uses: hadolint/hadolint-action@v2.1.0
+        with:
+          dockerfile: ${{ matrix.flavor }}/Dockerfile
+  build-and-push:
+    needs: [dockerfile-lint]
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        flavor:
+          - bullseye-slim
+          - scratch
+    steps:
+      - uses: actions/checkout@v3
+      - uses: docker/metadata-action@v4
+        id: meta
+        with:
+          images: skynetlabs/skyd
+          flavor: |
+            latest=${{ matrix.flavor == 'scratch' && 'auto' || 'false' }}
+            suffix=-${{ matrix.flavor }}
+          tags: |
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            type=semver,pattern={{major}}
+            type=semver,pattern=,suffix=${{ matrix.flavor }}
+            type=sha,enable=${{ github.event_name == 'pull_request' }}
+      - uses: docker/login-action@v2
+        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/')
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+      - uses: docker/setup-buildx-action@v2
+      - uses: docker/build-push-action@v3
+        with:
+          context: ${{ matrix.flavor }}
+          build-args: |
+            branch=${{ (github.event_name == 'pull_request' && 'master') || github.event.inputs.branch || github.ref_name }}
```

Is this the branch name of `skyd`'s repo or this repo?
```suggestion
            branch=${{ (github.event_name == 'pull_request' && 'main') || github.event.inputs.branch || github.ref_name }}
```

---

that's the source skyd branch so that should default to [master](https://gitlab.com/SkynetLabs/skyd/-/tree/master)

---

we should add a comment here to prevent someone from thinking this is a typo in the future.

---

documented in https://github.com/SkynetLabs/docker-skyd/pull/1/commits/65bf2f23c48a30892488050edb15ceea00094b5d